src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -684,8 +684,20 @@ public function dumpFile($filename, $content)
         // when the filesystem supports chmod.
         $tmpFile = $this->tempnam($dir, basename($filename));
 
-        if (false === @file_put_contents($tmpFile, $content)) {
-            throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
+        $expectedSize = false;
+
+        if (\is_resource($content)) {
+            $stat = fstat($content);
+            if (false !== $stat) {
```

Comment:
Sorry that's not generic enough: a userland stream wrapper could very well decode to return 0 ou -1 here. The only safe way would be to stream "manually".
BUT I would really like to know how PHP behaves in out of space situations here.
Could you give it a try? Mounting a loopback filesystem could be a way to try maybe? (I don't know how better than Google sorry).

Reply:
Good idea to check again! Here is an example:
```php -ddisplay_errors=1 -r 'var_dump(file_put_contents("another-file", str_repeat("a", 1024 * 1024)));'

Warning: file_put_contents(): Only 401408 of 1048576 bytes written, possibly out of free disk space in Command line code on line 1
bool(false)
```

It even leaves behind a file:
```
 ls -lh another-file
-rw-r--r-- 1 vagrant vagrant 392K Jun  9 19:14 another-file
```

If this is indeed cross-platform behavior than I misdiagnosed the issue here – in fact there wouldn't be an issue. The tempfile couldn’t be fully written and `file_put_contents()` returns false, the exception is thrown, everything is fine.

Reply:
Cool, thanks for double-checking. This means we can close as everything is fine: an error will happen as expected, isn't it?

Reply:
If that behavior is consistent across platform, yes. I don’t have access to Windows nor would I know how to simulate this. Do you have anybody in mind?

Reply:
It would make sense that this behaves the same on Windows.
I'm not able to verify. But let's close until someone proves there is something to do.
Thanks for having a look!